### DISCUSSION TITLE
DDS Router set up multi-hopping questions with Security?

### DISCUSSION THREAD

**Discussion by MrSodaCanDan** [eProsima/DDS-Router#432] [General]
Hello everyone, 
First Issue: I am new to DDS routing and looking to create a something similar to the image in  https://eprosima-dds-router.readthedocs.io/en/latest/rst/use_cases/repeater.html. I've been trying to combine multiple different example yaml files such as wan_client (NetA) and wan_server (NetB & NetC) with the repeater example to connect to my last NetworkA to NetworkB to NetworkC with robots on NetworkA and NetworkC. 
Second Issue: Looking to use SROS across each network sharing the same domain ID. How does one go about setting that up in DDS router so that they have the same files so the can see each other in the discovery process?

**Comment by rsanchez15** [comment]
Hi @MrSodaCanDan

I don't understand exactly what infrastructure you have in mind. A diagram with the nodes and the different networks you have would be helpful for us. 

In any case, [here](https://docs.vulcanexus.org/en/latest/rst/tutorials/cloud/cloud_tutorials.html) are the Vulcanexus Cloud tutorials. They explain how to use DDS Router with ROS 2 and provide multiple architectures and distributed deployments with ROS 2. In addition there is a specific tutorial on how to use [DDS Router with SROS](https://docs.vulcanexus.org/en/latest/rst/tutorials/cloud/secure_router/secure_router.html), explaining how to create and distribute keys and permissions to the different nodes so that they can communicate through DDS Router.

**Comment by MrSodaCanDan** [comment]
Hi Raul,

I attached an image of what we are looking at for a simple example. We are looking to create more of the edge to edge through multiple networks daisy chained together such as the repeater example (https://docs.vulcanexus.org/en/latest/rst/tutorials/cloud/edge_edge_repeater_cloud/edge_edge_repeater_cloud.html).  
The resources you sent me are great to start with and I appreciate the help so far! My question would be how can I integrate this example into one using security. Would it be just following step by step as the security you sent above?  Does security work between two WAN participants the same way as between two domains? 
[dds_topology (1).pdf](https://github.com/eProsima/DDS-Router/files/14335984/dds_topology.1.pdf)

**Comment by MrSodaCanDan** [comment]
This is updated version of trying to run the repeater example with my topology above. I tried to just run DDS-Router1 with a ros2 applications running on the same node. I am trying to connect through DDS-Router2 (repeater) to DDS-Router3 running an application on the same machine similar to https://docs.vulcanexus.org/en/latest/rst/tutorials/cloud/edge_edge_repeater_cloud/edge_edge_repeater_cloud.html. 
Here is the yaml file for DDS-ROUTER1:
_________________________________________
version: v4.0

participants:

  - name: ROS_2_LAN_1
    kind: local

  - name: Router_Client
    kind: wan
    connection-addresses:
      - ip: 192.168.4.2
        port: 12345
        transport: tcp
__________________________________________    

Here is the yaml file for DDS-ROUTER2 (repeater):
__________________________________________
version: v4.0

participants:

  - name: Router_Repeater
    kind: wan
    repeater: true
    listening-addresses:
      - ip: 192.168.4.2
        port: 12345
        transport: tcp
      - ip: 192.168.4.2
        port: 12346
        transport: tcp                                  
__________________________________________

Here is the yaml file for DDS-ROUTER3:
__________________________________________
version: v4.0

participants:

  - name: ROS_2_LAN_2
    kind: local

  - name: Router_Client
    kind: wan
    connection-addresses:
      - ip: 192.168.4.2
        port: 12346
        transport: tcp
__________________________________________

If anyone can tell me where I'm going wrong it would be much appreciated. The repeater node is is connected to both routes via static routes and can DDS-Router1 and DDS-Router3 can ping each other but nothing in ros is transfer correctly yet.

**Comment by MrSodaCanDan** [comment]
@rsanchez15 
Looking for any kind of help would be much appreciated.
